app/src/main/java/org/mozilla/tv/firefox/navigationoverlay/channels/ChannelTile.kt

```diff
@@ -23,7 +23,8 @@ data class ChannelTile(
     val title: String,
     val setImage: (ImageView) -> Unit,
     val tileSource: TileSource,
-    val id: String
+    val id: String,
+    val type: TileType
```

Comment:
This seems like it's the same concept as `TileSource`.  Could we merge these into one?

Reply:
For the record, I am worried about these.  We've had a lot of unnecessary branching logic based on tile type in the past.  Merging them into `ChannelTile` wasn't easy, and I'm worried that we could be setting ourselves up to repeat that mistake.